whats-on/app/views/pages/whats-on.njk

```diff
@@ -0,0 +1,91 @@
+{% extends 'layout/default.njk' %}
+
+{% set metaContent = {
+  type: 'website',
+  title: "What's On",
+  image: exhibitionAndEventPromos.temporaryExhibitionPromos[0].image.contentUrl,
+  url: pageConfig.canonicalUri,
+  description: ''
+} %}
+
+{% block pageMeta %}
+  {% component 'open-graph', { pageConfig: pageConfig, metaContent: metaContent } %}
+  {% component 'twitter-card', { pageConfig: pageConfig, metaContent: metaContent } %}
+  <meta name="description" content="{{ metaContent.description }}" />
+{% endblock %}
+
+{% block body %}
+   <script type="application/ld+json">
+    [{% for exhibition in exhibitionAndEventPromos.temporaryExhibitionPromos %}
+      {% set featuredImage = {contentUrl: exhibition.image.contentUrl} %}
+      {% set exhib = exhibition | objectAssign({safeDescription: exhibition.description | safe | striptags, featuredImage: featuredImage}) %}
+      {{ exhib | jsonLd('exhibitionLd') | safe }},
+    {% endfor %}
+    {% for event in exhibitionAndEventPromos.eventPromos %}
+      {{ event | jsonLd('eventPromoLd') | safe }},
+    {% endfor %}
+    {% for exhibition in exhibitionAndEventPromos.permanentExhibitionPromos %}
+      {% set featuredImage = {contentUrl: exhibition.image.contentUrl} %}
+      {% set exhib = exhibition | objectAssign({safeDescription: exhibition.description | safe | striptags, featuredImage: featuredImage}) %}
+      {{ exhib | jsonLd('exhibitionLd') | safe }}{{ ',' if not loop.last }}
+    {% endfor %}]
+  </script>
+
+  {% set todayDateString = exhibitionAndEventPromos.dates.today + '|' + exhibitionAndEventPromos.dates.today %}
+  {% set weekendDateString = exhibitionAndEventPromos.dates.weekend[0] + '|' + exhibitionAndEventPromos.dates.weekend[1] %}
+  {% set allDateString = exhibitionAndEventPromos.dates.all[0] + '|' + exhibitionAndEventPromos.dates.all[1] %}
+  {% set whatsOnModel = {
+    todayOpeningHours: todayOpeningHours,
+    name: 'What\'s on',
+    items: [
+      {id: 'today', title: 'Today', url: '/whats-on/?' + 'f[dates]' | urlencode + '=' + todayDateString | urlencode },
+      {id: 'weekend', title: 'This weekend', url: '/whats-on/?' + 'f[dates]' | urlencode + '=' + weekendDateString | urlencode },
+      {id: 'everything', title: 'Everything', url: '/whats-on/?' + 'f[dates]' | urlencode + '=' + allDateString | urlencode}
+    ]
+   } %}
```

Comment:
I wonder if this could all sit outside of the template too?

Reply:
I had a similar thought, but then forgot it